node/actors/control_api.go

```diff
@@ -181,16 +181,108 @@ func (api *ControlAPI) subscribe() error {
 }
 
 func (api *ControlAPI) handleAuction(m *nats.Msg) {
-	// TODO
+	req := new(nodecontrol.AuctionRequestJson)
+	err := json.Unmarshal(m.Data, req)
+	if err != nil {
+		api.logger.Error("Failed to unmarshal auction request", slog.Any("error", err))
+		respondEnvelope(m, AuctionResponseType, 500, nil, fmt.Sprintf("failed to unmarshal auction request: %s", err))
+		return
+	}
+
+	ctx := context.Background()
+	_, agentSuper, err := api.self.ActorSystem().ActorOf(ctx, AgentSupervisorActorName)
+	if err != nil {
+		api.logger.Error("Failed to locate agent supervisor actor", slog.Any("error", err))
+		respondEnvelope(m, AuctionResponseType, 500, nil, fmt.Sprintf("failed to locate agent supervisor actor: %s", err))
+		return
+	}
+
+	askResp, err := api.self.Ask(ctx, agentSuper, auctionRequestToProto(req))
+	if err != nil {
+		api.logger.Error("Failed to getdo auction", slog.Any("error", err))
+		respondEnvelope(m, AuctionResponseType, 500, nil, fmt.Sprintf("failed to do auction: %s", err))
+		return
+	}
+
+	protoResp, ok := askResp.(*actorproto.AuctionResponse)
+	if !ok {
+		api.logger.Error("Workload listing response from agent supervisor was not the correct type")
+		respondEnvelope(m, AuctionResponseType, 500, nil, "Agent supervisor returned the wrong data type")
+		return
+	}
+
+	respondEnvelope(m, AuctionResponseType, 200, auctionResponseFromProto(protoResp), "")
 }
 
 func (api *ControlAPI) handleDeploy(m *nats.Msg) {
-	// TODO
+	req := new(nodecontrol.StartWorkloadRequestJson)
+	err := json.Unmarshal(m.Data, req)
+	if err != nil {
+		api.logger.Error("Failed to unmarshal deploy request", slog.Any("error", err))
+		respondEnvelope(m, RunResponseType, 500, nil, fmt.Sprintf("failed to unmarshal deploy request: %s", err))
+		return
+	}
+
+	ctx := context.Background()
```

Comment:
here we don't need to talk to the agent supervisor. Since we have the workload type name on the deployment request, we can use `actorOf` to find the agent by name, and send the deploy request to it. The agent supervisor doesn't need to proxy that message.